components/public-api-server/pkg/server/server.go

```diff
@@ -37,7 +41,9 @@ func Start(logger *logrus.Entry, cfg Config) error {
 	return nil
 }
 
-func register(srv *baseserver.Server, cfg Config) error {
+func register(srv *baseserver.Server, cfg Config, registry *prometheus.Registry) error {
+	registry.MustRegister(proxy.ConnectionDurationHistogram)
```

Comment:
The registry here shouldn't really know about the metric, and once you make it private as suggested, this won't be possible. Move this into the Proxy package, as something like `proxy.RegisterMetrics(registry)`